validate/validate.go

```diff
@@ -817,6 +817,18 @@ func (v *Validator) CheckLinuxResources() (errs error) {
 		}
 	}
 
+	if r.BlockIO != nil && r.BlockIO.WeightDevice != nil {
+		for _, weightDevice := range r.BlockIO.WeightDevice {
+			if weightDevice.Weight == nil && weightDevice.LeafWeight == nil {
+				errs = multierror.Append(errs,
+					specerror.NewError(
+						specerror.BlkIOWeightOrLeafWeightExist,
+						fmt.Errorf("You MUST specify at least one of `weight` or `leafWeight` in a given entry"),
```

Comment:
This string is already covered by `BlkIOWeightOrLeafWeightExist`.  I think we want something like:

```go
for i, weightDevice := range r.BlockIO.WeightDevice {
    …
        fmt.Errorf("linux.resources.blockIO.weightDevice[%d] specifies neither weight nor leafWeight", i)
    …
}

Reply:
updated, thanks.